File: ios/ShoreKit/Sources/ShoreKit/Models/Club.swift
import Foundation
import GRDB

public struct Club: Codable, Identifiable, Sendable {
    public var id: UUID
    public var regionId: UUID
    public var name: String
    public var abbreviation: String

    public init(id: UUID, regionId: UUID, name: String, abbreviation: String) {
        self.id = id
        self.regionId = regionId
        self.name = name
        self.abbreviation = abbreviation
    }
}

extension Club: FetchableRecord, PersistableRecord {
    public static let databaseTableName = "clubs"
}

File: src/Tides.Core/Domain/Club.cs
namespace Tides.Core.Domain;

public class Club
{
    public Guid Id { get; private set; }
    public Guid RegionId { get; private set; }
    public string Name { get; private set; } = null!;
    public string Abbreviation { get; private set; } = null!;

    private Club() { }

    public Club(Guid id, Guid regionId, string name, string abbreviation)
    {
        Id = id;
        RegionId = regionId;
        Name = name;
        Abbreviation = abbreviation;
    }
}
